dbgd.c

```diff
@@ -298,3 +300,65 @@ static int dbgd_show_front_screen(Dbg__Request *req, Dbg__Response *res)
 
     return DBG__RESPONSE__TYPE__OK;
 }
+
+static int dbgd_call(Dbg__Request *req, Dbg__Response *res)
+{
+    if (!req->has_address)
+        return DBG__RESPONSE__TYPE__ERROR_INCOMPLETE_REQUEST;
+
+    // These variables will be used as parameters for inline assembly.
+    // We make them static as the stack pointer will be changed.
+    // So we want to address them globally.
+    static uint32_t stack_pointer;
+    static uint32_t stack_backup;
+    static uint32_t address;
+
+    // Allocate a stack for working and space for supplied data
+    size_t stack_size = 0x1000;
+    if (req->has_data) {
+      stack_size += req->data.len;
+    }
+    uint8_t* stack_data = malloc(stack_size);
+
+    // Push optional stack contents, starting at top of stack
+    stack_pointer = (uint32_t)&stack_data[stack_size];
+    if (req->has_data) {
+      stack_pointer -= req->data.len;
+      memcpy((void*)stack_pointer, req->data.data, req->data.len);
+    }
+
+    // Set address to call
+    address = req->address;
+
+    asm("pusha\n"
+        "mov %%esp, %[stack_backup]\n" // Keep copy of original stack
+
+        "mov %[stack_pointer], %%esp\n"
+
+        "call *%[address]\n" // Call the function
```

Comment:
Some functions do pass values through registers, so I can think of instances where we would want to support setting up registers before calling. We can do this with a proxy function I suppose, but it's a little messier. It would be easy to just have the client setup the registers on the stack then we can `popa` them into place before jumping to the function.

Reply:
If you review by commit, you'll find the original code which did that. However, I found it not beneficial in almost all cases, and more troublesome for the API.

*Bonus points if you can mention a single *useful* (for us) fastcall function on Xbox.*

Reply:
Yes, this is the same use-case I had in mind. If we have an existing XBE, it would be possible to call the title existing title or title library functions, some of which may use the `fastcall` convention. This is just another reason a more general purpose call operation is useful to have.